src/main/java/edu/unc/lib/boxc/migration/cdm/services/SourceFileService.java

```diff
@@ -427,8 +425,8 @@ private List<String> gatherFilesystemCandidatePaths(SourceFileMappingOptions opt
         Files.walkFileTree(basePath, new SimpleFileVisitor<>() {
             @Override
             public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) {
-                if (!Files.isDirectory(file) && options.getExtensions().contains(FilenameUtils.getExtension(file.toString()))
-                        && !origSourcePaths.contains(basePath.relativize(file).toString())) {
+                if (!Files.isDirectory(file) && !origSourcePaths.contains(FilenameUtils.getName(file.toString()))
```

Comment:
`FilenameUtils.getName(file.toString())` I'm not sure I understand this, isn't origSourcePaths populated with absolute paths, so we'd want to check for full file paths in there?

Reply:
origSourcePaths is populated with `originalRecord.get(2)`, which is `basePath.relativize(file).toString()`. I can populate origSourcePaths with absolute paths.